openapi-cli/src/test/java/io/ballerina/openapi/generators/service/ServiceContractTests.java

```diff
@@ -0,0 +1,169 @@
+
+    @Test(description = "Test default service object type name")
+    public void testDefaultServiceTypeNameInGeneratedService() throws IOException, BallerinaOpenApiException {
+        Path definitionPath = RES_DIR.resolve("swagger/service_type/default_service_type_name.yaml");
+        OpenAPI openAPI = GeneratorUtils.getOpenAPIFromOpenAPIV3Parser(definitionPath);
+        OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                .withOpenAPI(openAPI)
+                .withFilters(filter)
+                .build();
+        TypeHandler.createInstance(openAPI, false);
+        ServiceDeclarationGenerator ballerinaServiceGenerator = new ServiceDeclarationGenerator(oasServiceMetadata);
+        ServiceContractGenerator serviceContractGenerator = new ServiceContractGenerator(oasServiceMetadata,
+                ballerinaServiceGenerator.getFunctionsList());
+        syntaxTree = serviceContractGenerator.generateSyntaxTree();
+        CommonTestFunctions.compareGeneratedSyntaxTreewithExpectedSyntaxTree(
+                "service_type/default_service_type_name.bal", syntaxTree);
+    }
+
+    @Test(description = "Test custom service object type name")
+    public void testCustomServiceTypeNameInGeneratedService() throws IOException, BallerinaOpenApiException {
+        Path definitionPath = RES_DIR.resolve("swagger/service_type/custom_service_type_name.yaml");
+        OpenAPI openAPI = GeneratorUtils.getOpenAPIFromOpenAPIV3Parser(definitionPath);
+        OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                .withOpenAPI(openAPI)
+                .withFilters(filter)
+                .withServiceObjectTypeName("CustomServiceObjectTypeName")
+                .build();
+        TypeHandler.createInstance(openAPI, false);
+        ServiceDeclarationGenerator ballerinaServiceGenerator = new ServiceDeclarationGenerator(oasServiceMetadata);
+        ServiceContractGenerator serviceContractGenerator = new ServiceContractGenerator(oasServiceMetadata,
+                ballerinaServiceGenerator.getFunctionsList());
+        syntaxTree = serviceContractGenerator.generateSyntaxTree();
+        CommonTestFunctions.compareGeneratedSyntaxTreewithExpectedSyntaxTree(
+                "service_type/custom_service_type_name.bal", syntaxTree);
+    }
+
+    @Test(description = "Test custom service object type name with special characters")
+    public void testCustomServiceTypeNameWithSpecialCharacters() throws IOException, BallerinaOpenApiException {
+        Path definitionPath = RES_DIR.resolve("swagger/service_type/custom_name_with_special_characters.yaml");
+        OpenAPI openAPI = GeneratorUtils.getOpenAPIFromOpenAPIV3Parser(definitionPath);
+        OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                .withOpenAPI(openAPI)
+                .withFilters(filter)
+                .withServiceObjectTypeName("CustomServiceObject%$&TypeName1")
```

Comment:
What I meant is something like this `"Album-Service"` or `"Album Service"`. In the generated name I expect `Album\-Service` and `Album\ Service` rather than sanitizing it.

Reply:
This is resolved with the following commit:
[f3d4727f2e376de54b2250edcda1497dd444a497](https://github.com/ballerina-platform/openapi-tools/pull/1799/commits/f3d4727f2e376de54b2250edcda1497dd444a497)